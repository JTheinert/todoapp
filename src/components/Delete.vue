<template>
  <div class="dialog-item">
    <h1 class="delete-text">
      Are you sure you want to delete this Todo ?
    </h1>
    <button
      class="btn-style-yes"
      @click="removeTodo(nummer)">
      Yes
    </button>
    <button
      class="btn-style"
      @click="cancelDelete()">no</button>
  </div>
</template>

<script>
/* eslint-disable no-console */

export default {
  name: 'Delete',
  components: {},

  computed: {
    todos () {
      return this.$store.state.todos
    },
    nummer () {
      return this.$store.state.nummer
    }
  },
  methods: {
    removeTodo (nummer) {
      console.log(nummer)
      this.$store.state.todos.splice(nummer, 1)
      this.$router.push('/home')
    },
    removeTodoDetail (index) {
      this.todos.splice(index, 1)
    },
    editTodo (todo) {
      todo.editing = true
      this.beforeEditCache = todo.title
    },

    cancelDelete () {
      this.dialog = !this.dialog
      this.$router.push('/home')
    },
    openModal (index) {
      this.dialog = !this.dialog
      this.nummer = index
    },
    openModalDetails (index, todo) {
      this.dialog = !this.dialog
      todo.isOpen = !todo.isOpen
      todo.buttonName = todo.isOpen ? 'Close' : 'Details'
      this.nummer = index
    }
  }
}
</script>

<style>
.dialog-item {
display: block;
font-size: 30px;
text-align: center;
margin-top: 50px;
}
.delete-text {
display: block;
font-size: 30px;
border: 3px solid #2a9cac;
text-align: center;
border-radius: 10px;
background-color: #2a9cac;
color: white;
}

.btn-style {
background-color: #2a9cac;
font-size: 30px;
border-radius: 10px;
padding: 15px;
margin: 10px;
width: 150px;
height: 100%;
color: white;
}
.btn-style-yes {
background-color: red;
font-size: 30px;
border-radius: 10px;
padding: 15px;
margin: 10px;
width: 150px;
height: 100%;
color: white;
}

.delete-button {
background-color: red;
font-size: 20px;
border-radius: 10px;
text-align: center;
color: white;
}

.details-delete-button {
background-color: red;
font-size: 20px;
border-radius: 10px;
text-align: center;
color: white;
margin-left: 10px;
}

.done {
color: green;
}
.progress {
color: darkgoldenrod;
}

.image {
margin: 5px;
height: 40px;
}
.filter {
display: flex;
margin-left: 10px;
margin-right: 10px;
color: #2a9cac;
justify-content: space-around;
margin-bottom: 10px;
}
.teaser-text {
display: flex;
font-size: 15px;
text-align: left;
justify-content: left;
}
</style>
