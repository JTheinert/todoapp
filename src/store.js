/* eslint-disable no-console */
import Vue from 'vue'
import Vuex from 'vuex'

Vue.use(Vuex)

export const store = new Vuex.Store({
  mutations: {
    updateIsOpen (state, { id, isOpen }) {
      const todo = state.todos.find(obj => obj.id === id)
      todo.isOpen = isOpen
    },
    updateFilteredNr () {
      // eslint-disable-next-line no-use-before-define
      const filteredNr = filteredNr
    },

    set (newTodo) {
      return newTodo
    },
    updateNummer (state, { nummer }) {
      state.nummer = nummer
    },
    updateIdForTodo (state) {
      state.idForTodo++
    }
  },
  methods: {
    mounted () {
      // eslint-disable-next-line no-console
      console.log('App mounted!')
      if (localStorage.getItem('todos')) { this.todos = JSON.parse(localStorage.getItem('todos')) }
    },
    watch: {
      todos: {
        handler () {
          console.log('Todos changed!')
          localStorage.setItem('todos', JSON.stringify(this.todos))
        },
        deep: true
      }
    }
  },
  state: {
    filteredNr: 1,
    modalShow: false,
    newTodo: '',
    idForTodo: 1,
    beforeEditCache: '',
    beforeEditDescriptionCache: '',
    placeholder: 'Edit Description',
    dialog: false,
    nummer: 0,
    todos: [
      {
        id: 0,
        title: 'cool',
        editing: false,
        descriptionEditing: false,
        isOpen: false,
        status: "isn't done",
        description: '',
        filter: false,
        filterNr: 1,
        used: 3
      }
    ]
  },

  getters: {
    doneTodos: state => {
      return state.todos.filter(todo => todo.done)
    },
    doneTodosCount: (getters) => {
      return getters.doneTodos.length
    },
    getTodoById: state => id => {
      return state.todos.find(todo => todo.id === id)
    },
    getName: () => 'rajesh'
  }
})
