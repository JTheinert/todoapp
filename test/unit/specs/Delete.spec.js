import Delete from '@/src/Components/Delete.vue'
import { mount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'
import { store } from '@/store'
import VueRouter from 'vue-router'

const localVue = createLocalVue()
localVue.use(Vuex)

describe('Delete.vue', () => {
  const wrapper = mount(Delete, {store, VueRouter})

  test('function has been called', () => {
    const wrapper = mount(Delete, {store, VueRouter})

    expect(wrapper.text()).toContain('no')
  })

  test('Delete todo Button', () => {
    const removeTodo = jest.fn()

    wrapper.setMethods({
      removeTodo: removeTodo
    })

    wrapper.find('button.btn-style-yes').trigger('click')

    expect(removeTodo).toHaveBeenCalled()
  })

  test('close modal', () => {
    const cancelDelete = jest.fn()

    wrapper.setMethods({
      cancelDelete: cancelDelete
    })

    wrapper.find('button.btn-style').trigger('click')

    expect(cancelDelete).toHaveBeenCalled()
  })
  test('renders correctly', () => {
    const wrapper = mount(Delete)
    expect(wrapper.element).toMatchSnapshot()
  })
})
