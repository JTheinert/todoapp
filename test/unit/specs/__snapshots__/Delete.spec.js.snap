// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Delete.vue renders correctly 1`] = `
<div
  class="dialog-item"
>
  <h1
    class="delete-text"
  >
    
    Are you sure you want to delete this Todo ?
  
  </h1>
   
  <button
    class="btn-style-yes"
  >
    
    Yes
  
  </button>
   
  <button
    class="btn-style"
  >
    no
  </button>
</div>
`;
